### ISSUE TITLE
Show tail of the execution log if run fails

### ISSUE THREAD

**Issue by psss** [psss/tmt#82]
If there is an error encountered during execution the last 30 lines (or so) of stderr/stdout should be displayed on the output to make it easier to debug the problem. This should be enabled by default for the debug level. In verbose and info level it would be nice to always give a reasonable message about the failed command. Or, perhaps, the tail output should be enabled always? @thrix, @pvalena, what do you think?

**Comment by thrix** [comment]
I wonder, wouldn't be better if we would take the approach which gluetool is taking with --output option and have the debug output be always written out to a debug log in `run` directory and when the command fails reference this log to the user where he can do additional investigation?

I agree that if we can, let's print reasonable user errors.

**Comment by pvalena** [comment]
I strongly agree with outputing the errors occured (we can put 10 lines there and path to the file containing the log). I'd rather avoid chaning the log location based on user request (`--output`). That's what `--id` is for, right?

Personally, I prefer catching the expected errors, and even parsing the error output and recovering / producing meaningful error message. Traceback (log) is the last resort.

**Comment by pvalena** [comment]
F.e.:
 - https://github.com/psss/tmt/blob/master/tmt/steps/provision/vagrant.py#L218
 - https://github.com/psss/tmt/blob/master/tmt/steps/prepare/__init__.py#L137
It would be nice if we'd have that implemented in some generalal way.

**Comment by psss** [comment]
Good ideas! Makes sense to me. To sum up:

* Always include detailed debug output under the `run` directory
* Whenever possible try to detect problems and provide a reasonable error
* Point users to the log file to investigate details of the error

Does this look good to you?

**Comment by psss** [comment]
This has been fixed by #218. Example output:

```
> tmt run -a plan --name smoke prepare -h shell -s 'sudo dnf install -y tmttt'
/var/tmp/tmt/run-413

/plans/smoke
    discover
        how: shell
        directory: /home/psss/git/tmt
        summary: 1 test selected
    provision
        how: local
        distro: Fedora release 31 (Thirty One)
        kernel: 5.5.11-200.fc31.x86_64
        summary: 1 guest provisioned
    prepare
        how: shell
        overview: 1 script found
    finish

stdout (2/2 lines)
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Last metadata expiration check: 0:13:15 ago on St 29. dubna 2020, 14:37:52 CEST.
No match for argument: tmttt

stderr (1/1 lines)
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Error: Unable to find a match: tmttt

 ERROR  Failed to run command 'sudo dnf install -y tmttt'.
```
